tests/nnstreamer_source/unittest_src_iio.cc

```diff
@@ -1564,6 +1564,96 @@ TEST (test_tensor_src_iio, data_verify_freq_generic_type)
   clean_iio_dev_structure (dev0);
 }
 
+/**
+ * @brief test the unusual/exceptional cases.
+ */
+TEST (test_tensor_src_iio, unusual_cases)
+{
+  iio_dev_dir_struct *dev0;
+  GstElement *src_iio_pipeline;
+  GstElement *src_iio;
+  GstStateChangeReturn status;
+  GstState state;
+  gchar *parse_launch;
+  gint samp_freq;
+  gint data_value;
+  guint data_bits;
+  gchar *ret_device;
+  gchar *ret_trigger;
+  gchar *iio_base_dir;
+  gchar *dev_dir;
+  gchar *ret_channels;
+
+  data_value = DATA;
+  data_bits = 16;
+  gint samp_freq_idx = 1;
+  /** Make device */
+  dev0 = make_full_device (data_value, data_bits);
+  ASSERT_NE (dev0, nullptr);
+  /** setup */
+  samp_freq = (gint) g_ascii_strtoll (samp_freq_avail[samp_freq_idx], NULL, 10);
+  dev0->log_file = g_build_filename (dev0->base_dir, "temp.log", NULL);
+  parse_launch =
+      g_strdup_printf
+      ("%s iio-base-dir=%s dev-dir=%s device-number=%d trigger-number=%d silent=FALSE frequency=%d "
+      "merge-channels-data=False name=my-src-iio channels=1,3,5 ! multifilesink location=%s",
+      ELEMENT_NAME, dev0->iio_base_dir_sim, dev0->dev_dir, 0, 0, samp_freq, dev0->log_file);
+  src_iio_pipeline = gst_parse_launch (parse_launch, NULL);
+
```

Comment:
Need `g_free (parse_launch)` here

Reply:
it's added :)

Reply:
I think 'g_free (parse_launch)' should be located here. Because it can be dangerous to free after assert (line 1605).